File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        unordered_map<char,int> freq;
        int left=0;
        int ans=0;
        int length=0;
        for(int right=0;right<s.length();right++){
            
            freq[s[right]]++;
            while(freq[s[right]]>2){
                freq[s[left]]--;
                left++;
                
            }
            length=right-left+1;
            ans=max(ans,length);
        }
        return ans;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int left=0;
        int ans=0;
        HashMap<Character,Integer> freq=new HashMap<>();
        for(int right=0;right<s.length();right++){
            char ch=s.charAt(right);
            freq.merge(ch,1,Integer::sum);
            while(freq.get(ch)>2){
                char ch1=s.charAt(left);
                freq.put(ch1,freq.get(ch1)-1);
                left++;
            }
            int length=right-left+1;
            ans=Math.max(ans,length);
        }
        return ans;

    }
}